src/routes/v2/project_creation.rs

```diff
@@ -965,3 +990,284 @@ async fn process_icon_upload(
         Err(CreateError::InvalidIconFormat(file_extension.to_string()))
     }
 }
+
+#[derive(Serialize, Deserialize, Validate, Clone)]
+struct CollectionCreateData {
+    #[validate(
+        length(min = 3, max = 64),
+        custom(function = "crate::util::validate::validate_name")
+    )]
+    #[serde(alias = "collection_name")]
+    /// The title or name of the project.
+    pub title: String,
+    #[validate(
+        length(min = 3, max = 64),
+        regex = "crate::util::validate::RE_URL_SAFE"
+    )]
+    #[serde(alias = "collection_slug")]
+    /// The slug of a collection, used for vanity URLs
+    pub slug: String,
+    #[validate(length(min = 3, max = 255))]
+    #[serde(alias = "collection_description")]
+    /// A short description of the collection.
+    pub description: String,
+    #[validate(length(max = 65536))]
+    #[serde(alias = "collection_body")]
+    /// A long description of the collection, in markdown.
+    pub body: String,
+
+    #[validate(length(max = 32))]
+    #[serde(default = "Vec::new")]
+    /// A list of initial projects to use with the created collection
+    pub initial_projects: Vec<ProjectId>,
+
+    /// If the collection should initialize to public
+    pub public: bool,
+}
+#[post("collection")]
+pub async fn collection_create(
+    req: HttpRequest,
+    mut payload: Multipart,
+    client: Data<PgPool>,
+    redis: Data<deadpool_redis::Pool>,
+    file_host: Data<Arc<dyn FileHost + Send + Sync>>,
+    session_queue: Data<AuthQueue>,
+) -> Result<HttpResponse, CreateError> {
+    let mut transaction = client.begin().await?;
+    let mut uploaded_files = Vec::new();
+
+    let result = collection_create_inner(
+        req,
+        &mut payload,
+        &mut transaction,
+        &***file_host,
+        &mut uploaded_files,
+        &client,
+        &redis,
+        &session_queue,
+    )
+    .await;
+
+    if result.is_err() {
+        let undo_result = undo_uploads(&***file_host, &uploaded_files).await;
+        let rollback_result = transaction.rollback().await;
+
+        undo_result?;
+        if let Err(e) = rollback_result {
+            return Err(e.into());
+        }
+    } else {
+        transaction.commit().await?;
+    }
+
+    result
+}
+
+#[allow(clippy::too_many_arguments)]
+pub async fn collection_create_inner(
+    req: HttpRequest,
+    payload: &mut Multipart,
+    transaction: &mut sqlx::Transaction<'_, sqlx::Postgres>,
+    file_host: &dyn FileHost,
+    uploaded_files: &mut Vec<UploadedFile>,
+    pool: &PgPool,
+    redis: &deadpool_redis::Pool,
+    session_queue: &AuthQueue,
```

Comment:
this doesn't need to be a multipart route, this can just be a route like report creation which just takes a json body. users can add icons later